token-swap/js/cli/token-swap-test.js

```diff
@@ -325,6 +328,72 @@ export async function withdraw(): Promise<void> {
   assert(info.amount.toNumber() == tokenB);
 }
 
+export async function swapOneTransaction(): Promise<void> {
+  console.log('Creating swap token a account');
+  let userAccountA = await mintA.createAccount(owner.publicKey);
+  await mintA.mintTo(userAccountA, owner, [], SWAP_AMOUNT_IN);
+
+  const balanceNeeded = await Token.getMinBalanceRentForExemptAccount(
+    connection,
+  );
+  const newAccount = new Account();
```

Comment:
For my own understanding, does this create a new keypair? In this function, it looks like `newAccount` gets passed in as a signer. If so, does the secret key for `newAccount` have to be stored so it can be reused later?

Reply:
My guess is that it's required for `SystemProgram.createAccount` but isn't required afterwards since `owner` has control over it. Feels weird to generate a private key and discard it afterwards, but this kind of makes sense.

Reply:
That's exactly right -- the code is really simple over at https://github.com/solana-labs/solana/blob/master/web3.js/src/account.js#L22
And yes, since `newAccount` is your new token B account, you will need access to at least its public key.  Otherwise, since it's a `Token` account, it's actually owned by the `Token` program, and `owner` has the right to issue certain instructions to the `Token` program to make changes to the account.

Reply:
Got it. Thanks for the explanation :+1: